disvae/losses.py

```diff
@@ -240,6 +264,170 @@ def __call__(self, data, model, optimizer, storer):
 
         return vae_loss
 
+class BatchTCLoss(BaseLoss):
+    """
+        Compute the decomposed KL loss with either minibatch weighted sampling or
+        minibatch stratified sampling according to [1]
+
+        Parameters
+        ----------
+        alpha : float
+            Weight of the mutual information term.
+
+        beta : float
+            Weight of the total correlation term.
+
+        gamma : float
+            Weight of the dimension-wise KL term.
+
+        latent_dim: int
+            Dimension of the latent variable
+
+        mss : bool
+            Selects either minibatch stratified sampling (True) or minibatch
+            weighted  sampling (False)
+
+        References :
+           [1] Chen, Tian Qi, et al. "Isolating sources of disentanglement in variational
+           autoencoders." Advances in Neural Information Processing Systems. 2018.
+    """
+
+    def __init__(self, alpha, beta, gamma, data_size, latent_dim, mss=False):
```

Comment:
`is_mss` for consistency. Default hyperparameters for `alpha`, `beta`, `gamma` should be given. 

I don't see any place you actually need `latent_dim` better drop it for having similar constructor to the other losses. Everywhere I see `latent_dim` you can actually get it from the shapes of the input vectors (seem like it, I might be wrong)

Reply:
I have changed both of these.